azafea/event_processors/endless/metrics/v3/model/views.py

```diff
@@ -0,0 +1,678 @@
+    __materialized__ = True
+
+    __query__ = DbSession().query(
+        ProgramDumpedCoreV2.occured_at,
+        OSVersion.version.label('os_version'),
+        ProgramDumpedCoreV2.info,
+        Machine.image_id,
+        Machine.location.label('site'),
+        Machine.dualboot.label('dual_boot'),
+        Machine.live
+    ).select_from(
+        ProgramDumpedCoreV2
+    ).outerjoin(
+        Request
+    ).outerjoin(
+        Machine, Request.machine_id == Machine.machine_id
+    ).outerjoin(
+        OSVersion
+    ).filter(
+        Machine.demo.is_(False)
+    ).distinct(
+        ProgramDumpedCoreV2.id
+    )
+
+
+class UpdaterFailureViewV2(View):
+    __tablename__ = 'updater_failure_view_v2'
+    __materialized__ = True
+
+    __query__ = DbSession().query(
+        UpdaterFailureV2.occured_at,
+        OSVersion.version.label('os_version'),
+        UpdaterFailureV2.component,
+        UpdaterFailureV2.error_message,
+        Machine.image_id,
+        Machine.location.label('site'),
+        Machine.dualboot.label('dual_boot'),
+        Machine.live
+    ).select_from(
+        UpdaterFailureV2
+    ).outerjoin(
+        Request
+    ).outerjoin(
+        Machine, Request.machine_id == Machine.machine_id
+    ).outerjoin(
+        OSVersion
+    ).filter(
+        Machine.demo.is_(False)
+    ).distinct(
+        UpdaterFailureV2.id
+    )
+
+
+class WindowsAppOpenedViewV2(View):
+    __tablename__ = 'windows_app_opened_view_v2'
+    __materialized__ = True
+
+    __query__ = DbSession().query(
+        WindowsAppOpenedV2.occured_at,
+        OSVersion.version.label('os_version'),
+        WindowsAppOpenedV2.argv,
+        Machine.image_id,
+        Machine.location.label('site'),
+        Machine.dualboot.label('dual_boot'),
+        Machine.live
+    ).select_from(
+        WindowsAppOpenedV2
+    ).outerjoin(
+        Request
+    ).outerjoin(
+        Machine, Request.machine_id == Machine.machine_id
+    ).outerjoin(
+        OSVersion
+    ).filter(
+        Machine.demo.is_(False)
+    ).distinct(
+        WindowsAppOpenedV2.id
+    )
+
+
+class StartupFinishedViewV2(View):
+    __tablename__ = 'startup_finished_view_v2'
+    __materialized__ = True
+
+    __query__ = DbSession().query(
+        StartupFinishedV2.occured_at,
+        OSVersion.version.label('os_version'),
+        StartupFinishedV2.firmware,
+        StartupFinishedV2.loader,
+        StartupFinishedV2.kernel,
+        StartupFinishedV2.initrd,
+        StartupFinishedV2.userspace,
+        StartupFinishedV2.total,
+        Machine.image_id,
+        Machine.location.label('site'),
+        Machine.dualboot.label('dual_boot'),
+        Machine.live
+    ).select_from(
+        StartupFinishedV2
+    ).outerjoin(
+        Request
+    ).outerjoin(
+        Machine, Request.machine_id == Machine.machine_id
+    ).outerjoin(
+        OSVersion
+    ).filter(
+        Machine.demo.is_(False)
+    ).distinct(
+        StartupFinishedV2.id
+    )
+
+
```

Comment:
I guess these are just not yet implemented but this branch is missing the 2 × 3 tables to turn old sequence events into new aggregate events.

Reply:
They’re not created yet, that’s right.